corehq/apps/reports/filters/api.py

```diff
@@ -247,3 +249,30 @@ def paginate_options(data_sources, query, start, size):
         size -= len(objects)  # how many more do we need for this page?
         options.extend(objects)
     return total, options
+
+
+class DeviceLogFilter(LoginAndDomainMixin, JSONResponseMixin, View):
+    field = None
+
+    def get(self, request, domain):
+        q = self.request.GET.get('q', None)
+        field_filter = {self.field + "__startswith": q}
```

Comment:
Couldn't think of anything more performant that would still look through the device entry log as those aren't in ES. Could use full text search if we had jdjango 1.10 (https://docs.djangoproject.com/en/1.10/ref/contrib/postgres/search/)